setup.py

```diff
@@ -107,7 +107,6 @@ def initialize_options(self):
         f"numpy{get_version_spec('numpy')}",
         f"iree-base-compiler{get_version_spec('iree-base-compiler')}",
         f"iree-base-runtime{get_version_spec('iree-base-runtime')}",
-        "torch>=2.3.0",
```

Comment:
Let's update the docs too.

https://github.com/iree-org/iree-turbine?tab=readme-ov-file#quick-start-for-users
https://github.com/iree-org/iree-turbine?tab=readme-ov-file#install-pytorch-for-your-system

Specifically mention that installing `torch` is the user's responsibility and drop these comments:

> The above does install some CUDA/cuDNN packages which are unnecessary for most usage.

> If no explicit action is taken, the default PyTorch version will be installed.

Reply:
In the IREE docs we can add a Linux/macOS/Windows branch in the docs here too: https://iree.dev/guides/ml-frameworks/pytorch/#prerequisites. I'll send a PR for that.

Reply:
> In the IREE docs we can add a Linux/macOS/Windows branch in the docs here too: https://iree.dev/guides/ml-frameworks/pytorch/#prerequisites. I'll send a PR for that.

Thanks! Updated the docs here, excuse me initialy missing this.

Reply:
Sent https://github.com/iree-org/iree/pull/19493 for https://iree.dev/guides/ml-frameworks/pytorch/#prerequisites